sphinx/domains/c.py

```diff
@@ -3126,6 +3137,7 @@ class CObject(ObjectDescription[ASTDeclaration]):
     option_spec: OptionSpec = {
         'noindexentry': directives.flag,
         'nocontentsentry': directives.flag,
+        'singlelinesig': directives.flag,
```

Comment:
This needs to be solved in a better way, but I'd prefer not to compound the readability problem by adding another flag in the same style. 
```suggestion
        'single-line-signature': directives.flag,
```